R/mutate_mzroll_list.R

```diff
@@ -50,12 +50,95 @@ floor_peaks <- function(mzroll_list,
       )),
     missing_peaks
   )
-
+  
   mzroll_list$measurements <- completed_peaks
+  
+  return(mzroll_list)
+}
 
+#' Impute missing peaks with provided feature-level imputation values
+#'
+#'@param mzroll_list: data in triple omic structure
+#'@param lod_values: a tibble that maps groupId to log2 feature-level imputation values
+#'@param quant_var: column to use for peak values
+#'@param imputation_sd: standard deviation of Gaussian distribution to use for missing peak imputation
+#'
+#'@return triple omic data with imputed missing peaks
+#'
+#' @examples
+#' mzroll_list_imputed <- impute_missing_peaks(mzroll_list, lod_values, "log2_abundance", 0.15)
+#'
+#'@export
+impute_missing_peaks <- function(mzroll_list,
+                                 lod_values,
+                                 quant_var = "log2_abundance",
+                                 imputation_sd = 0.15) {
+  
+  test_mzroll_list(mzroll_list)
+  
+  valid_quant_var <- setdiff(
+    mzroll_list$design$measurements$variable,
+    c(mzroll_list$design$feature_pk, mzroll_list$design$sample_pk)
+  )
+  
+  checkmate::assertChoice(quant_var, valid_quant_var)
+  checkmate::assertNumeric(mzroll_list$measurements[[quant_var]])
+  
+  ## find missing peaks
+  missing_peaks <- tidyr::expand_grid(
+    groupId = mzroll_list$features$groupId,
+    sampleId = mzroll_list$samples$sampleId) %>%
+    dplyr::anti_join (
+      mzroll_list$measurements,
+      by = c("groupId", "sampleId")
+    )
+  
+  ## impute missing peaks
+  missing_peaks_imputed <- left_join(
+    missing_peaks, 
+    group_background, 
+    by = c("groupId")) %>%
+    rowwise() %>%
+    mutate(!!rlang::sym(quant_var) := rnorm(1, mean = !!rlang::sym(quant_var)+1, sd = imputation_sd))
+  
+  ## merge measured peaks with imputed peaks
+  completed_peaks <- dplyr::bind_rows(
+    mzroll_list$measurements,
+    missing_peaks_imputed
+  )
+  
+  mzroll_list$measurements <- completed_peaks
   return(mzroll_list)
 }
 
+#' Fill in missing peaks
+#'
```

Comment:
Thanks for refactoring this way!
Maybe consider adding an `@description` section clarifying that depending on the inputs provided, this will call `impute_missing_peaks()` or `floor_peaks()`. You can reference these in code formating using `\code` tag, e.g.

'@description
'If \code{fill_values} is a data frame, this function calls \code{impute_missing_peaks()}. If it is a numeric vector, 'this function calls \code{floor_peaks()}. Other types are not currently supported.

Reply:
Thanks @PMSeitzer added the description.